patches/tModLoader/Terraria.ModLoader.IO/TagSerializable.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Reflection;
+
+namespace Terraria.ModLoader.IO
+{
+	public interface TagSerializable
+	{
+		TagCompound SerializeData();
+	}
+
+	public interface ITagDeserializer<out T> where T : TagSerializable
```

Comment:
Thinking about it now I agree, I only made it like this because I was kind of "inspired" by how Android has it done with its Parcel class and Parcelable interface.